### PR TITLE
Unit test refactoring

### PR DESCRIPTION
**Pull request by jtaubensee**
## Description
This PR improves the unit tests with the following:
* Disabled parallelization, and moved logging functionality to TestUtility.
* Added a test attribute that will log the beginning/end of each test along with the current platform
* Added permutations for authenticated/unauthenticated entities. Now nearly all tests run against authenticated/unauthenticated endpoints.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

**Comment by jtaubensee** [comment]
`Write1MB` was renamed to `WriteLargeDataSetTest`, but still appears to be flaky.

**Comment by jtaubensee** [comment]
@dstuckims - Can you please review?

File: test/Microsoft.Azure.Relay.UnitTests/DisplayTestNameAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    using System.Reflection;
+    using Microsoft.Extensions.PlatformAbstractions;
+    using Xunit.Sdk;
+
+    public class DisplayTestMethodNameAttribute : BeforeAfterTestAttribute
+    {
+        public override void Before(MethodInfo methodUnderTest)
+        {
+            TestUtility.Log($"Begin {methodUnderTest.DeclaringType}.{methodUnderTest.Name} on {PlatformServices.Default.Application.RuntimeFramework.ToString()}");
```
Comment (dstuckims) [inline test/Microsoft.Azure.Relay.UnitTests/DisplayTestNameAttribute.cs]: >.ToString() [](start = 148, length = 11)

Nit: Calling .ToString() here isn't strictly necessary.

File: test/Microsoft.Azure.Relay.UnitTests/HybridConnectionTestBase.cs [hunk 2]
```diff
@@ -4,39 +4,260 @@
...
         {
-            this.Logger = new Logger(output);
+            Authenticated,
+            Unauthenticated
+        }
+
+        public static IEnumerable<object> AuthenticationTestPermutations => new object[]
+        {
+            new object[] { EndpointTestType.Authenticated },
+            new object[] { EndpointTestType.Unauthenticated }
```
Comment (dstuckims) [inline test/Microsoft.Azure.Relay.UnitTests/HybridConnectionTestBase.cs]: Do these truly need to be object arrays inside the outer object array?  I guess maybe that's how Xunit does it so one can pass multiple params to the test method.

File: test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs [hunk 3]
```diff
@@ -5,222 +5,132 @@ namespace Microsoft.Azure.Relay.UnitTests
...
     public class RunTimeTests : HybridConnectionTestBase
     {
-        const string TestAcceptHandlerResultHeader = "X-TestAcceptHandlerResult";
-        const string TestAcceptHandlerStatusCodeHeader = "X-TestAcceptHandlerStatusCode";
-        const string TestAcceptHandlerStatusDescriptionHeader = "X-TestAcceptHandlerStatusDescription";
-        const string TestAcceptHandlerSetResponseHeader = "X-TestAcceptHandlerSetResponseHeader";
-        const string TestAcceptHandlerDelayHeader = "X-TestAcceptHandlerDelay";
-
-        public RunTimeTests(ITestOutputHelper output) : base(output)
+        public RunTimeTests() : base()
```
Comment (dstuckims) [inline test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs]: > : base() [](start = 29, length = 9)

Nit: " : base()" text is optional.  The same behavior occurs even if omitted.

File: test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs [hunk 4]
```diff
@@ -237,202 +147,201 @@ public RunTimeTests(ITestOutputHelper output) : base(output)
...
                 listener = new HybridConnectionListener(fakeEndpointConnectionString);
                 var client = new HybridConnectionClient(fakeEndpointConnectionString);
 
-                await Assert.ThrowsAsync<RelayException>(() => listener.OpenAsync(TimeSpan.FromSeconds(30)));
+#if NET451
+                await Assert.ThrowsAsync<EndpointNotFoundException>(() => listener.OpenAsync());
+                await Assert.ThrowsAsync<EndpointNotFoundException>(() => client.CreateConnectionAsync());
+#else
+                await Assert.ThrowsAsync<RelayException>(() => listener.OpenAsync());
                 await Assert.ThrowsAsync<RelayException>(() => client.CreateConnectionAsync());
```
Comment (dstuckims) [inline test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs]: Consider adding a "TODO" mentioning this is due to a bug in .NET Core (maybe even reference the issue#).

File: test/Microsoft.Azure.Relay.UnitTests/WebSocketTests.cs [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+    public class WebSocketTests : HybridConnectionTestBase
+    {
+        const string TestAcceptHandlerResultHeader = "X-TestAcceptHandlerResult";
+        const string TestAcceptHandlerStatusCodeHeader = "X-TestAcceptHandlerStatusCode";
+        const string TestAcceptHandlerStatusDescriptionHeader = "X-TestAcceptHandlerStatusDescription";
+        const string TestAcceptHandlerSetResponseHeader = "X-TestAcceptHandlerSetResponseHeader";
+        const string TestAcceptHandlerDelayHeader = "X-TestAcceptHandlerDelay";
+
+        public  WebSocketTests() : base()
```
Comment (dstuckims) [inline test/Microsoft.Azure.Relay.UnitTests/WebSocketTests.cs]: > : base() [](start = 32, length = 9)

Nit: Again " : base()" is optional.  (My general guideline is "If optional text doesn't change anything from the language default then don't add that text".  Or maybe "if the optional text doesn't change anything from a *well-known* language default...".  In general for me, the fewer characters my eyes and brain have to see, the better :)

**Comment by dstuckims** [comment]
:shipit:

File: test/Microsoft.Azure.Relay.UnitTests/HybridConnectionTestBase.cs (hunk 2 above)
Comment (jtaubensee) [inline test/Microsoft.Azure.Relay.UnitTests/HybridConnectionTestBase.cs]: To my knowledge, yes. I was unable to get them working any other way.

**Pull request merged**